From 3121249d0536f4398fc154e3c5511d3dcaf33ce5 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 19 Nov 2012 17:10:54 +0200
Subject: [PORT FROM MAIN] Telephony: Fix wrong hangup request sent

BZ: 5267

If an incoming call is received while the ongoing call is
disconnected, incoming call is accepted automatically.
This is a corner case.

Hangup of the active call when there is no background call
is resulting in RIL_REQUEST_HANGUP_FOREGROUND_RESUME_BACKGROUND
request instead of RIL_REQUEST_HANGUP request. Due to this
incoming call is accepted automatically.

Change-Id: I536b0a32760893bd5d0c054efb012ad26ca44170
Orig-Change-Id: Iae796f055b6f3c3b04d494fd951ad853a74d58d5
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/75572
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Lucas, Guillaume <guillaume.lucas@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../internal/telephony/gsm/GsmCallTracker.java     |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
index f866078..46fda83 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
@@ -752,6 +752,8 @@ public final class GsmCallTracker extends CallTracker {
                     log("(foregnd) hangup dialing or alerting...");
                 }
                 hangup((GsmConnection)(call.getConnections().get(0)));
+            } else if (mBackgroundCall.isIdle()) {
+                hangupAllConnections(call);
             } else {
                 hangupForegroundResumeBackground();
             }
-- 
1.7.4.1

